onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidateName.kt

```diff
@@ -0,0 +1,24 @@
+package com.alexafit.onboardingauthdomain.useCase
+
+import com.alexafit.onboardingauthdata.local.util.namevalidator.NameValidator
+import com.alexafit.onboardingauthdomain.model.domain.mapper.ValidateNameResult
+import javax.inject.Inject
+
+class ValidateName @Inject constructor(
+    private val nameValidator: NameValidator
+) {
+    operator fun invoke(name: String): ValidateNameResult {
+        val isValidName = nameValidator.isValidName(name)
+        return if (name.isBlank()) {
+            ValidateNameResult(
+                userName = name.trim(),
+                isValidName
+            )
+        } else {
+            ValidateNameResult(
+                userName = name,
+                isValidName
+            )
+        }
+    }
+}
```

Comment:
The `nameValidator` is pretty useless here, since there's nothing that really makes sense to abstract (like an SDK call that doesn't belong in domain). For the email validator the `Patterns.EMAIL_ADDRESS` call comes from the Android SDK which shouldn't be in domain, but for both name and passwords you only need simple logic to validate these, so that can all be in the use case